runtime/queries/dart/locals.scm

```diff
@@ -0,0 +1,20 @@
+; Scopes
+;-------
+
+[
+ (block)
+ (try_statement)
+ (catch_clause)
+ (finally_clause)
+] @local.scope
+
+; Definitions
+;------------
+
+(class_definition
+ body: (_) @local.definition)
```

Comment:
So, you will probably want to add more `@local.definition` queries to get this to work correctly, this should be used for variable definitions:

```
var a = 1 <-- this is a @local.definition

something(
  a // <-- this is a @local.reference
)
```

but `locals.scm` isn't that important, so if you want I can merge this PR as is and we fix this later on?

Reply:
I did rebase and force-push to fix the languages.toml. For this part I think it might be best to get the main support in, and then myself and others will continue to build it up while using it. I don't fully understand how tree-sitter works yet and have limited time to dive deeper sorry.

Reply:
Sounds good!